File: src/main/java/my_salary/classes/ShiftsInfo.java
package my_salary.classes;

import lombok.Data;

import java.time.LocalTime;

@Data
public class ShiftsInfo {
    public Float eveningPercentage;
    public Float nightPercentage;
    public LocalTime morningStartHour;
    public LocalTime eveningStartHour;
    public LocalTime nightStartHour;

    @Override
    public String toString() {
        return "ShiftsInfo{" + "eveningPercentage=" + eveningPercentage + ", nightPercentage=" + nightPercentage + ", morningStartHour=" + morningStartHour + ", eveningStartHour=" + eveningStartHour + ", nightStartHour=" + nightStartHour + '}';
    }
}

File: client/lib/Model/ShiftsInfo.dart
import 'package:intl/intl.dart';

class ShiftsInfo {
  double? eveningPercentage;
  double? nightPercentage;
  DateTime? morningStartHour;
  DateTime? eveningStartHour;
  DateTime? nightStartHour;

  ShiftsInfo({
    this.eveningPercentage,
    this.nightPercentage,
    this.morningStartHour,
    this.eveningStartHour,
    this.nightStartHour,
  });

  factory ShiftsInfo.fromJson(Map<String, dynamic> json) {
    DateFormat format = DateFormat("HH:mm:ss");
    return ShiftsInfo(
      eveningPercentage: (json['eveningPercentage'] as num?)?.toDouble(),
      nightPercentage: (json['nightPercentage'] as num?)?.toDouble(),
      morningStartHour: format.parse(json['morningStartHour']),
      eveningStartHour: format.parse(json['eveningStartHour']),
      nightStartHour: format.parse(json['nightStartHour']),
    );
  }

  Map<String, dynamic> toJson() {
    DateFormat format = DateFormat("HH:mm:ss");
    return {
      'eveningPercentage': eveningPercentage,
      'nightPercentage': nightPercentage,
      'morningStartHour': morningStartHour != null ? format.format(morningStartHour!) : "00:00:00",
      'eveningStartHour': eveningStartHour != null ? format.format(eveningStartHour!) : "00:00:00",
      'nightStartHour': nightPercentage != null ? format.format(nightStartHour!) : "00:00:00",
    };
  }
}